docs/universal-gateway/examples/front-door-pattern.mdx

```diff
@@ -0,0 +1,94 @@
+---
+
+import ReserveDomain from "./snippets/_reserve-domain.mdx";
+import CloudEndpoint from "./snippets/_cloud-endpoint.mdx";
+import Back from "./snippets/_back-to-examples.mdx";
+
+The "front door" pattern involves using a Cloud Endpoint, the forward-internal Traffic Policy action, and an internal Agent Endpoint to route traffic from the public internet to your service(s).
+
+With this pattern, you can:
+
+- Host any number of services under a single hostname (e.g. `your-company.com`).
+- Decouple gateway configuration from per-service traffic management policies.
+- Prevent any driect exposure of your services or their host systems to the public internet.
+- Route traffic and add authentication to microservices with a single API gateway configuration.
+- Authenticate developer environments or CI preview builds behind a single endpoint.
+
+## 1. Create an endpoint for your service
+
+Start an internal [Agent Endpoint](/docs/universal-gateway/agent-endpoints/), replacing `$PORT` based on where your service listens.
+You can also use one of our [SDKs](/docs/agent/sdks) or the [Kubernetes Operator](/docs/k8s).
+
+```bash
+ngrok http $PORT --url https://service.internal
+```
+
+## 2. Reserve a domain
+
+<ReserveDomain />
+
+## 3. Create a Cloud Endpoint
+
+<CloudEndpoint />
+
+## 4. Add routing to your service with Traffic Policy
+
+Copy and paste the policy below into your new Cloud Endpoint in the ngrok dashboard.
+
+```yaml
+on_http_request:
+  - actions:
+      - type: forward-internal
+        config:
+          url: http://service.internal
+```
+
+### What's happening here?
+
+On every HTTP request, the Traffic Policy engine forwards the traffic to the
+internal agent endpoint you created at `https://service.internal`.
+
+## 5. Add a second service and routing (optional)
+
+If you have another service to host under the front door pattern, start another agent.
+
+```bash
+ngrok http $PORT --url https://service-two.internal
+```
+
+## 5a. Update your routing based on paths
+
+Copy and paste the policy below into the ngrok dashboard, replacing `/one` and `/two` with the paths you'd like to use for routing traffic to each service.
+
+```yaml
+on_http_request:
+  - expressions:
+      - "req.url.includes('/one')"
+    actions:
+      - type: forward-internal
+        config:
+          url: http://service.internal
+  - expressions:
+      - "req.url.includes('/two')"
+    actions:
+      - type: forward-internal
+        config:
+          url: http://service-two.internal
+```
+
+## 6. Try out your endpoint!
+
+- In your browser: https://your-service.ngrok.app
+- In your terminal: `curl https://your-service.ngrok.app`
+
+## What's next?
+
+- Read more about [Traffic Policy](/docs/traffic-policy) and its [core concepts](/docs/traffic-policy/concepts).
+- Add authentication to your endpoint with the [`oauth` action](/docs/traffic-policy/actions/oauth) action or [`basic-auth` action](/docs/traffic-policy/actions/basic-auth).
```

Comment:
```suggestion
- [Traffic Policy](/docs/traffic-policy) enables you to do many more things worth your endpoints. For example, try adding authentication to your endpoint with the [`oauth` action](/docs/traffic-policy/actions/oauth) or [`basic-auth` action](/docs/traffic-policy/actions/basic-auth) actions.
```
I think we should condense these two into one point because just telling people to learn more about Traffic Policy doesn't give them a reason why, and so they'll be hesitant to click that link. But the second point was basically explaining why you should use traffic policy, so it's really one point.